### PR TITLE
perf(rust, python): improve performance for datetime parsing with %Z

### PR DESCRIPTION
**Pull request by MarcoGorelli**
Currently, `'%Z'` is either ignored within `strptime::parse`, or it just causes parsing to fallback to `NaiveDateTime::parse_from_str`

I'd suggest to just exclude it from `strptime::fmt_len` and let it go directly to the chrono parser

This comes with a little [perf improvement](https://www.kaggle.com/code/marcogorelli/polars-timing?scriptVersionId=121211658) for parsing local time zone names with `'%Z'`
```python
dates = pl.date_range(datetime(1, 1, 1), datetime(9999, 1, 1)).dt.strftime('%Y-%m-%d %H:%MUTC')

dates.str.strptime(pl.Datetime, '%Y-%m-%d %H:%M%Z')
```

Here:
- 3.02 s ± 24 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
- 3.02 s ± 42.5 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
- 3.01 s ± 38 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

upstream/master:
- 3.4 s ± 324 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
- 3.24 s ± 39.2 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)
- 3.33 s ± 57.4 ms per loop (mean ± std. dev. of 7 runs, 1 loop each)

### CODE REVIEW

File: polars/polars-time/src/chunkedarray/utf8/mod.rs [hunk 1]
```diff
@@ -482,7 +482,6 @@ pub trait Utf8Methods: AsUtf8 {
                         // Safety:
                         // fmt_len is correct, it was computed with this `fmt` str.
                         unsafe { self::strptime::parse(s.as_bytes(), fmt.as_bytes(), fmt_len) }
-                            .or_else(|| NaiveDateTime::parse_from_str(s, &fmt).ok())
```
Comment (MarcoGorelli) [inline polars/polars-time/src/chunkedarray/utf8/mod.rs:485]: anything which `fmt_len` returns `Some` for should be parseable by `strptime::parse`, so it should be OK to remove this fallback

File: polars/polars-time/src/chunkedarray/utf8/strptime.rs [hunk 2]
```diff
@@ -50,10 +50,17 @@ pub(super) fn compile_fmt(fmt: &str) -> String {
...
 pub(super) unsafe fn parse(val: &[u8], fmt: &[u8], fmt_len: u16) -> Option<NaiveDateTime> {
-    const ESCAPE: u8 = b'%';
-    if val.len() < fmt_len as usize {
+    let mut offset = 0;
+    let mut negative = false;
+    if val.starts_with(b"-") && fmt.starts_with(b"%Y") {
+        offset = 1;
+        negative = true;
+    }
+    if val.len() - offset != (fmt_len as usize) {
```
Comment (MarcoGorelli) [inline polars/polars-time/src/chunkedarray/utf8/strptime.rs:59]: having removed `'%Z'`, all other directives are fixed-length, so this check can be made more precise

File: py-polars/tests/unit/datatypes/test_temporal.py [hunk 3]
```diff
@@ -2166,6 +2166,15 @@ def test_tz_aware_with_timezone_directive(
     assert result == expected
 
 
+def test_local_time_zone_name() -> None:
+    ser = pl.Series(["2020-01-01 03:00ACST"]).str.strptime(
+        pl.Datetime, "%Y-%m-%d %H:%M%Z"
+    )
+    result = ser[0]
+    expected = datetime(2020, 1, 1, 3)
+    assert result == expected
```
Comment (MarcoGorelli) [inline py-polars/tests/unit/datatypes/test_temporal.py:2175]: this already passed on `master`, but there was no test with `'%Z'` parsing a local time zone name

**Comment by ritchie46** [comment]
Nice!

**Pull request merged**